tests/e2e/uninstall.go

```diff
@@ -0,0 +1,194 @@
+		By("Verifying Zookeeper-operator helm chart resources cleanup")
+
+		k8sResourceKinds, err := listK8sResourceKinds(kubectlOptions, "")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		zookeeperAvailableResourceKinds := stringSlicesInstersect(zookeeperCRDs(), k8sResourceKinds)
+		zookeeperAvailableResourceKinds = append(zookeeperAvailableResourceKinds, basicK8sResourceKinds()...)
+
+		remainedResources, err := getK8sResources(kubectlOptions,
+			zookeeperAvailableResourceKinds,
+			fmt.Sprintf(managedByHelmLabelTemplate, zookeeperOperatorHelmDescriptor.ReleaseName),
+			"",
+			kubectlArgGoTemplateKindNameNamespace,
+			"--all-namespaces")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(remainedResources).Should(BeEmpty())
+	})
+}
+
+// requireRemoveZookeeperOperatorCRDs deletes the zookeeper-operator CRDs
+func requireRemoveZookeeperOperatorCRDs(kubectlOptions k8s.KubectlOptions) {
+	It("Removing zookeeper-operator CRDs", func() {
+		for _, crd := range zookeeperCRDs() {
+			deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
+		}
+	})
+}
+
+// requireUninstallingPrometheusOperator uninstall prometheus-operator Helm chart and
+// remove CRDs.
+func requireUninstallingPrometheusOperator(kubectlOptions k8s.KubectlOptions) {
+	When("Uninstalling prometheus-operator", func() {
+		requireUninstallingPrometheusOperatorHelmChart(kubectlOptions)
+		requireRemovePrometheusOperatorCRDs(kubectlOptions)
+	})
+}
+
+// requireUninstallingPrometheusOperatorHelmChart uninstall prometheus-operator Helm chart
+// and checks the success of that operation.
+func requireUninstallingPrometheusOperatorHelmChart(kubectlOptions k8s.KubectlOptions) {
+	It("Uninstalling Prometheus-operator Helm chart", func() {
+		uninstallHelmChartIfExists(kubectlOptions, "prometheus-operator", true)
+		By("Verifying Prometheus-operator helm chart resources cleanup")
+
+		k8sResourceKinds, err := listK8sResourceKinds(kubectlOptions, "")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		prometheusAvailableResourceKinds := stringSlicesInstersect(prometheusCRDs(), k8sResourceKinds)
+		prometheusAvailableResourceKinds = append(prometheusAvailableResourceKinds, basicK8sResourceKinds()...)
+
+		remainedResources, err := getK8sResources(kubectlOptions,
+			prometheusAvailableResourceKinds,
+			fmt.Sprintf(managedByHelmLabelTemplate, prometheusOperatorHelmDescriptor.ReleaseName),
+			"",
+			kubectlArgGoTemplateKindNameNamespace,
+			"--all-namespaces")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(remainedResources).Should(BeEmpty())
+	})
+}
+
+// requireRemovePrometheusOperatorCRDs deletes the Prometheus-operator CRDs
+func requireRemovePrometheusOperatorCRDs(kubectlOptions k8s.KubectlOptions) {
+	It("Removing prometheus-operator CRDs", func() {
+		for _, crd := range prometheusCRDs() {
+			deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
```

Comment:
**Required**: We **MUST** check potential errors here.

```suggestion
			err := deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
			
			Expect(err).NotTo(HaveOccurred())
```

Reply:
fixed: [3efc5f9](https://github.com/banzaicloud/koperator/pull/987/commits/3efc5f92e73093a477784196d6bcf64ffcc0a417)